cmd/utils/auth.go

```diff
@@ -32,3 +46,69 @@ func GetCredentials() (*azidentity.ChainedTokenCredential, error) {
 
 	return chain, nil
 }
+
+// cachedInteractiveBrowserCredential is a credential that uses the interactive browser to authenticate and caches the token.
+// TODO: This is a workaround until the azidentity package supports caching, https://github.com/Azure/azure-sdk-for-go/issues/16643.
+type cachedInteractiveBrowserCredential struct {
+	client public.Client
+}
+
+func newCachedInteractiveBrowserCredential() (*cachedInteractiveBrowserCredential, error) {
+	file := path.Join(homedir.HomeDir(), ".kubectl-az", "token-cache.json")
```

Comment:
Did you test it on Windows? I suppose you use `k8s.io/client-go/util/homedir` to resolve the problem of different OS, right?

Reply:
Thanks for pointing it out, `k8s.io/client-go/util/homedir` is doing `kubeconfig` specific stuff. I updated to use `os.UserHomeDir()` and tested it works fine on windows as well!